elytron/WFCORE-5279-elytron-ssl-subsystem.adoc

```diff
@@ -0,0 +1,196 @@
+----
+
+* The same operation could be executed in Elytron TLS as follows:
++
+[source]
+----
+/subsystem=elytron-tls/server-ssl-context=httpsSSC:add(key-manager=httpsKM,protocols=["TLSv1.2"])
+----
+
+==== Wildfly Elytron
+
+SSL code in Elytron is currently dependent on security domains, and therefore the rest of the subsystem.
+To minimize dependencies in Elytron TLS, the SSL code will be split into 2 modules:
+
+* ``ssl-base``, composed of domain-agnostic Elytron components, as well as refactored classes to create an SSL context without the need for a security domain. This will provide capabilities for Elytron TLS.
+* ``ssl-domain``, including the current domain-dependent Elytron classes. This will be used by Elytron proper, in addition to ``ssl-base`` to enable configurations that require a security domain.
+
+A Maven artifact for ``ssl-base`` will also need to be created and uploaded to repositories.
+
+==== Galleon layers
+
+In order to make the most out of the proposed subsystem, appropriate Galleon layers for replacing the 
+full Elytron subsystem will be added. The point of this is to reduce the memory overhead of simple deployments
+as much as possible.
+
+==== Covered use-case scenarios
+
+* Standard one-way TLS (trust store with certificates for client side, key store and keypair for server side)
+* Basic configuration - specify TLS version, cipher suites, etc.
+* KeyStore- (truststore) and credential store-related configuration
+* Mutual (two-way) TLS (key store and trust store on both sides)
+* FIPS
+* Ability to add providers
+* Quick certificate setup with Let's Encrypt
+** Integration with Certificate Authorities
+
+A more complete list is being developed, feel free to comment any other possible use-cases.
+
+
+=== Nice-to-Have Requirements
+
+* user-friendly operations - up for consideration, this may be unnecessary as a lot is covered by elytron and elytron-tool
+
+=== Non-Requirements
+
+* Terminology in Elytron is split between "SSL" and "TLS"; it would be nice to simplify with one term going forward, and maintaining the other for compatibility.
+** Would affect Elytron, Elytron TLS, WildFly Core, and direct references in documentation
+
+=== Implementation Plan
+////
+Delete if not needed. The intent is if you have a complex feature which can 
+not be delivered all in one go to suggest the strategy. If your feature falls 
+into this category, please mention the Release Coordinators on the pull 
+request so they are aware.
+////
+
+
+The Undertow subsystem uses reference to an SSLContext resource provided with the 'org.wildfly.security.ssl-context' capability.
+This makes implementation of a SSL subsystem straight-forward - we need to create a   SSLContext resource with the same capability and we can wire it in by referencing its name.
+
+Unlike the 'elytron' subsystem we want the configuration as simple as possible - ideally one CLI command.
```

Comment:
One command is the only simplification for users I can see in this RFE now, the rest seems be rather a code refactoring on background. Having two subsystems seems rather confusing. Are there another visible advantages? Is there a way to keep just one subsystem? (For example a Galleon layer with just TLS related configuration in the Elytron subsystem.)

Reply:
Still need to updating this section from #371. Ideally, we'd also include an updated `elytron-tool.sh` for the improvements. I'll keep this in mind as I update the section

Reply:
One thing I wanted to add from some of the earlier planning discussions is that by splitting this into it's own subsystem, it reduces the number of dependencies needed, like in non-Galleon setups, so it reduces the overall server demand.